backend/authentication-microservice/src/main/java/com/ss/utopia/auth/controller/SessionController.java

```diff
@@ -7,38 +7,92 @@
-				.orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
-		response.addCookie(sessionCookie);
-		return ResponseEntity.status(HttpStatus.CREATED).build();
-	}
+  @Autowired
+  private UserService userService;
+  @Autowired
+  private SessionCookieProvider sessionCookieProvider;
+
+  /**
+   * Login a user using loginDto
+   *
+   * @param loginDto : { username, password }
+   * @return
+   */
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  public void login(@RequestBody @Valid LoginDto loginDto, HttpServletResponse response) {
+    final User user = userService.loadAuthenticatedUser(loginDto.getUsername(), loginDto.getPassword())
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    final Cookie sessionCookie = sessionCookieProvider.createSessionCookie(user)
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    response.addCookie(sessionCookie);
+  }
+
+  /**
+   * Login only an admin user using loginDto
+   *
+   * @param loginDto : { username, password }
+   * @return
+   */
+  @PostMapping("/admin")
+  @ResponseStatus(HttpStatus.CREATED)
+  public void loginAdmin(@RequestBody @Valid LoginDto loginDto, HttpServletResponse response) {
+    final User user = userService.loadAuthenticatedUser(loginDto.getUsername(), loginDto.getPassword())
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    // if user is not admin
+    if (!userService.isUserAdmin(user))
+      throw new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed");
+    final Cookie sessionCookie = sessionCookieProvider.createSessionCookie(user)
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    response.addCookie(sessionCookie);
+  }
+
+  /**
+   * Login only an agent user using loginDto
+   *
+   * @param loginDto : { username, password }
+   * @return
+   */
+  @PostMapping("/agent")
+  @ResponseStatus(HttpStatus.CREATED)
+  public void loginAgent(@RequestBody @Valid LoginDto loginDto, HttpServletResponse response) {
+    final User user = userService.loadAuthenticatedUser(loginDto.getUsername(), loginDto.getPassword())
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    // if user is not agent
+    if (!userService.isUserAgent(user))
+      throw new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed");
+    final Cookie sessionCookie = sessionCookieProvider.createSessionCookie(user)
+      .orElseThrow(() -> new HttpServerErrorException(HttpStatus.UNAUTHORIZED, "Login Failed"));
+    response.addCookie(sessionCookie);
+  }
+
+  @GetMapping
+  public User getSession(@AuthenticationPrincipal UserDetails currentUser) {
+    if (currentUser == null)
+      throw new HttpServerErrorException(HttpStatus.FORBIDDEN, "Unauthorized User");
```

Comment:
Right. I made it to return null. The response object returns a User. Isn't what we agreed on?

Reply:
Oh, I see. the response return {user:user}; Is it alright with you to change the API definitions to just return user. It makes the front-end a little bit cleaner without the extra wrapper?

Reply:
sounds good to me. we can talk about it more. Just set it up that way so that it would be easy to add more data to the session in the future if need be. I think all you would have to do is have the /api/session get return a SessionDto object that has a user property. for now we can leave it.